addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,472 @@
+        if message:
+            action = {
+                "view_mode": "form",
+                "type": "ir.actions.act_window",
+                "res_id": partner.id,
+                "res_model": "res.partner",
+                "views": [[self.env.ref("base.view_partner_form").id, "form"]],
+            }
+            error_msg = '\n'.join(map(str, message))
+            raise RedirectWarning(_("Please enter required details for %s \n" + error_msg, partner.display_name),
+                action, _('Go to Configuration')
+            )
+
+    def format_timestamp(self, value):
+        return value.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+
+    @api.model
+    def _l10n_pk_round_value(self, amount, precision_digits=2):
+        """
+            This method is call for rounding.
+            If anything is wrong with rounding then we quick fix in method
+        """
+        return round(amount, precision_digits) or 0.0  # avoid -0.0
+
+    def _l10n_pk_edi_extract_digits(self, string):
+        if not string:
+            return string
+        matches = re.findall(r"\d+", string)
+        result = "".join(matches)
+        return result
+
+    def _l10n_pk_is_global_discount(self, line):
+        return not line.tax_ids and line.price_subtotal < 0 or False
+
+    def _get_l10n_pk_edi_invoice_type(self, move_type):
+        return {
+            "in_invoice": 1,
+            "out_invoice": 2,
+            "in_refund": 3,
+            "out_refund": 4,
+        }.get(move_type)
+
+    def _get_l10n_pk_edi_fbr_ref_number(self, move):
+        if move.move_type in ['in_refund', 'out_refund']:
+            return move.reversed_entry_id._get_l10n_pk_edi_response_json().get("result")
+        return False
+
+    def _get_l10n_pk_edi_sro_serial_no(self, product_id):
+        return product_id.l10n_pk_edi_schedule_code.schedule_code[-3:]
+
+    @api.model
+    def _get_l10n_pk_edi_partner_address(self, partner):
+        zip_digits = self._l10n_pk_edi_extract_digits(partner.zip)
+        address_parts = [
+            partner.street or "",
+            partner.street2 or "",
+            partner.city or "",
+            partner.state_id.name or "",
+            f"Pin: {zip_digits}" if zip_digits else ""
+        ]
+        partner_address = ", ".join(part for part in address_parts if part)
+        return partner_address
+
+    @api.model
+    def _get_l10n_pk_edi_partner_vat(self, move):
+        company = move.company_id
+        partner = move.company_id.partner_id
+        action = {
+            "view_mode": "form",
+            "type": "ir.actions.act_window",
+            "res_id": company.id,
+            "res_model": "res.company",
+            "views": [[self.env.ref("base.view_company_form").id, "form"]],
+        }
+        if not partner.vat:
+            raise RedirectWarning(_(
+                "Please enter a NTN number for %s", company.partner_id.name),
+                action,
+                _('Go to Company')
+            )
```

Comment:
```suggestion
```
I guess one error message is enough that user has not set/invalid NTN on the company